sklearn/svm/_classes.py

```diff
@@ -242,6 +262,10 @@ def fit(self, X, y, sample_weight=None):
         self : object
             An instance of the estimator.
         """
+        self._validate_params()
+
+        check_random_state(self.random_state)
+
         if self.C < 0:
             raise ValueError("Penalty term must be positive; got (C=%r)" % self.C)
```

Comment:
The idea is that with this new centralized validation scheme you can remove any old existing manual validation so long as it is simple like this. If it is more complex, e.g. depends on _the data_ or _other params_ then it can be left in, but this kind of stuff should be removed.
```suggestion
```